### PR TITLE
Add unit tests for codegen.makeNativeMemberFunction

### PR DESCRIPTION
**Pull request by pcardune**
Pretty self explanatory. Only non-test changes are small refactoring of syntax for es6 goodness.

### CODE REVIEW

File: apps/src/codegen.js [hunk 1]
```diff
@@ -205,7 +205,7 @@ function marshalNativeToInterpreterObject(
     }
     interpreter.setProperty(retVal,
                             prop,
-                            exports.marshalNativeToInterpreter(interpreter,
+                            marshalNativeToInterpreter(interpreter,
                                                                value,
                                                                nativeObject,
                                                                maxDepth));
```
Comment (islemaster) [inline apps/src/codegen.js]: These aligned arguments are no longer aligned.  It'd be nice to reformat these.

File: apps/src/codegen.js [hunk 2]
```diff
@@ -347,16 +347,18 @@ exports.marshalInterpreterToNative = function (interpreter, interpreterVar) {
  * @param {Object} opts Options block with interpreter and maxDepth provided
  * @param {function} callback The interpreter supplied callback function
  */
-var createNativeCallbackForAsyncFunction = function (opts, callback) {
-  return function (nativeValue) {
+function createNativeCallbackForAsyncFunction(opts, callback) {
+  return (nativeValue) => {
```
Comment (islemaster) [inline apps/src/codegen.js]: Nit: Unneeded parens.

File: apps/src/codegen.js [hunk 3]
```diff
@@ -405,41 +407,57 @@ exports.createNativeInterpreterCallback = function (opts, intFunc) {
...
 /**
  * Generate a native function wrapper for use with the JS interpreter.
+ * @param {Object} opts - configuration options. See below.
+ * @param {boolean} opts.dontMarshal -
+ * @param {} nativeFunc -
+ * @param {} nativeParentObj -
+ * @param {} interpreter -
+ * @param {} maxDepth -
+ * @param {} nativeIsAsync -
+ * @param {} nativeCallsBackInterpreter -
```
Comment (islemaster) [inline apps/src/codegen.js]: This looks like you meant to go back and fill it in, then forgot.

Comment (pcardune) [reply apps/src/codegen.js]: yep... that is exactly what that means.

**Pull request merged**